Playbooks/AS-PagerDuty-Integration/azuredeploy.json

```diff
@@ -0,0 +1,392 @@
+                                        "severity": "@variables('EventSeverity')",
+                                        "source": "@triggerBody()?['workspaceInfo']?['ResourceGroupName']",
+                                        "summary": "@triggerBody()?['object']?['properties']?['title']",
+                                        "timestamp": "@triggerBody()?['object']?['properties']?['createdTimeUtc']"
+                                    },
+                                    "routing_key": "@variables('IntegrationKey')"
+                                },
+                                "headers": {
+                                    "Accept": "application/vnd.pagerduty+json;version=2",
+                                    "Content-Type": "application/json"
+                                },
+                                "method": "POST",
+                                "uri": "https://events.pagerduty.com/v2/enqueue"
+                            }
+                        },
+                        "Initialize_Entities": {
+                            "runAfter": {
+                                "Initialize_EventSeverity": [
+                                    "Succeeded"
+                                ]
+                            },
+                            "type": "InitializeVariable",
+                            "inputs": {
+                                "variables": [
+                                    {
+                                        "name": "Entities",
+                                        "type": "string"
+                                    }
+                                ]
+                            }
+                        },
+                        "Initialize_EventSeverity": {
+                            "runAfter": {
+                                "Initialize_IntegrationKey": [
+                                    "Succeeded"
+                                ]
+                            },
+                            "type": "InitializeVariable",
+                            "inputs": {
+                                "variables": [
+                                    {
+                                        "name": "EventSeverity",
+                                        "type": "string"
+                                    }
+                                ]
+                            }
+                        },
+                        "Initialize_IntegrationKey": {
+                            "runAfter": {},
+                            "type": "InitializeVariable",
+                            "inputs": {
+                                "variables": [
+                                    {
+                                        "name": "IntegrationKey",
+                                        "type": "string",
+                                        "value":  "[parameters('IntegrationKey')]"

```

Comment:
Converting **IntegrationKey** from a parameter to variable is not required. This would rather expose the secret string in clear text.

Reply:
Understood. This is resolved.